DistroLauncher/WindowsUserInfo.cpp

```diff
@@ -0,0 +1,112 @@
+//
+//    Copyright: 2021, Canonical Ltd.
+//  License: GPL-3
+//  This program is free software : you can redistribute itand /or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License.
+//  .
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+//  GNU General Public License for more details.
+//  .
+//  You should have received a copy of the GNU General Public License
+//  along with this program.If not, see < https://www.gnu.org/licenses/>.
+//  .
+//  On Debian systems, the complete text of the GNU General
+//  Public License version 3 can be found in "/usr/share/common-licenses/GPL-3".
+//
+
+// Holds together the user information retrieved from Win32 API's.
+
+#include "stdafx.h"
+
+namespace DistributionInfo {
+	// Keep this part private to this translation unit because I don't think they need to be exported.
+	namespace {
+		struct WindowsUserInfo {
+			std::wstring userName;
+			std::wstring realName;
+			std::wstring localeName;
+
+			// This member-function hand-codes the YAML generation to avoid 
+			// adding a lib just because of such small feature, which would be an overkill.
+			// Shall the need for more YAML manipulation in the DistroLauncher arise,
+			// this function should be changed to use a proper YAML manipulation library, such as yaml-cpp.
+			std::wstring toYaml() {
+				std::wstring fullYaml;
+
+				if (!localeName.empty()) {
+					fullYaml += L"Welcome:\n  lang: " + localeName;
+				}
+
+				if (!(realName.empty() && userName.empty())) {
+					fullYaml += L"\nWSLIdentity:";
```

Comment:
So, if we don’t have a local, the file will start with:
```

WSLIdentity:
```

1. I think you tested, but subiquity has no issue with this? (Standard yaml parsing shouldn’t, be better to be safe than sorry)
2. I think we can still inverse the logic:
  a. add \n to the end of localeName
  b. then starting directly with `WSLIdentity`

The file will then ends with a trailing newline, but this is already what you are doing if we have a realName or an userName. Also, there is currently no trailing newline if you only have a locale and no real/username.